vpr/src/tileable_rr_graph/tileable_rr_graph_gsb.cpp

```diff
@@ -1460,3 +1717,318 @@ void build_direct_connections_for_one_gsb(const RRGraphView& rr_graph,
     /* Build actual edges */
     rr_graph_builder.build_edges(true);
 }
+
+/* Vib edge builder */
+t_vib_map build_vib_map(const RRGraphView& rr_graph,
+                        const DeviceGrid& grids,
+                        const vtr::NdMatrix<const VibInf*, 3>& vib_grid,
+                        const RRGSB& rr_gsb,
+                        const std::vector<t_segment_inf>& segment_inf,
+                        const size_t& layer,
+                        const vtr::Point<size_t>& gsb_coordinate,
+                        const vtr::Point<size_t>& actual_coordinate,
+                        std::vector<std::vector<std::vector<std::map<std::string, size_t>>>> medium_mux_name2medium_index) {
+    VTR_ASSERT(rr_gsb.get_x() == gsb_coordinate.x() && rr_gsb.get_y() == gsb_coordinate.y());
+    
+    t_vib_map vib_map;
+
+    const VibInf* vib = vib_grid[layer][actual_coordinate.x()][actual_coordinate.y()];
+    auto phy_type = grids.get_physical_type({(int)actual_coordinate.x(), (int)actual_coordinate.y(), (int)layer});
+    VTR_ASSERT(!strcmp(vib->get_pbtype_name().c_str(), phy_type->name));
+    const std::vector<t_first_stage_mux_inf> first_stages = vib->get_first_stages();
+    for (size_t i_first_stage = 0; i_first_stage < first_stages.size(); i_first_stage++) {
+        std::vector<t_from_or_to_inf> froms = first_stages[i_first_stage].froms;
+        RRNodeId to_node = rr_graph.node_lookup().find_node(layer, actual_coordinate.x(), actual_coordinate.y(), MEDIUM, i_first_stage);
```

Comment:
Need to check if the to_node and from_node is part of the RRGSB. Otherwise, OpenFPGA cannot support the architecture.

Reply:
The MEDIUM nodes will be added into the RRGSB for checking if the specific nodes are in the RRGSB or not.

Reply:
It occurs to me that all the ipins and opins of a CLB connect to the VIB at its right-top side in our new VIB architecture. It causes that the pins in the left or the bottom side of the CLB is not part of the RRGSB. How to handle it?
<img width="1088" alt="GSB_VIB" src="https://github.com/user-attachments/assets/aec1b9f5-849a-445d-aa22-5a278a111be9">